testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AppInitiatedActionTotpSetupTest.java

```diff
@@ -116,12 +117,13 @@ public void setupTotpRegister() {
         totpPage.configure(totp.generateTOTP(totpPage.getTotpSecret()));
 
         events.poll(); // skip to totp event
-        String authSessionId = events.expectRequiredAction(EventType.UPDATE_TOTP).user(userId).detail(Details.USERNAME, "setuptotp").assertEvent()
+        String expectedString = keycloakUsingProviderWithId(UserProvider.class, "jpa") ? "setuptotp" : "setupTotp";
+        String authSessionId = events.expectRequiredAction(EventType.UPDATE_TOTP).user(userId).detail(Details.USERNAME, expectedString).assertEvent()
                 .getDetails().get(Details.CODE_ID);
 
         assertKcActionStatus(SUCCESS);
         
-        events.expectLogin().user(userId).session(authSessionId).detail(Details.USERNAME, "setuptotp").assertEvent();
+        events.expectLogin().user(userId).session(authSessionId).detail(Details.USERNAME, expectedString).assertEvent();
```

Comment:
Could it be the following instead?
```suggestion
        events.expectLogin().user(userId).session(authSessionId).detail(Details.USERNAME, expectedString).usernameIgnoringLettercase().assertEvent();
```